libraries/extractor/src/main/java/androidx/media3/extractor/ts/DtsReader.java

```diff
@@ -66,9 +97,13 @@ public final class DtsReader implements ElementaryStreamReader {
    * @param language Track language.
    */
   public DtsReader(@Nullable String language) {
-    headerScratchBytes = new ParsableByteArray(new byte[HEADER_SIZE]);
+    int headerSize = max(EXTSS_HEADER_SIZE_MAX, FTOC_MAX_HEADER_SIZE);
+    headerScratchBytes = new ParsableByteArray(new byte[headerSize]);
     state = STATE_FINDING_SYNC;
     timeUs = C.TIME_UNSET;
+    uhdAudioChunkId = new AtomicInteger();
```

Comment:
Shouldn't this `uhdAudioChunkId` also `reset` in the `seek` method below? I tried resetting it but then the tests fails with an error. 

When seeking, the last key frame wouldn't be the same and as this parameter stores the audio chunk ID of the last key frame, I would expect this to be cleared. What do you think?

Reply:
@rohitjoins, It seems to be good to reset the _uhdChunkId_ parameter during the seek operation.

I have added this line of code in the seek() function.
`uhdAudioChunkId.set(0);`
I did not see any issues when I tested using a streaming use case. I did seeking the player multiple times from the UI.

Could you share the details of the code changes you added and the tests you did?

Reply:
@rahulmohan,

I did not make any code changes, just added `uhdAudioChunkId.set(0);` in the `seek()` method and ran the `TsExtractorTest` tests. 5 tests fails for `sampleWithDtsUhd` test.

Reply:
@rohitjoins 
The issue was caused by the wrong UHD header size value and this resulted in a negative index calculation when the uhdAudioChunkId is zero. I have fixed this issue in [b8f0350](https://github.com/androidx/media/pull/275/commits/b8f0350ddbb50c6d74b758c99d40d9d674c61009)

It should be fine to use the uhdAudioChunkId value from the last sync frame. So, no need to reset the uhdAudioChunkId in the seek() function.

Reply:
In Table 6-20 (ETSI TS 103 491 V1.2.1 (2019-05)), you can see that the m_unAudioChunkID is extracted from the navigation table. 

```
// Get the audio chunk ID from the stored list
unAudioChunkID =
m_CAudioChunkNaviLits.m_pCAudioChunksNaviParams[unListIndex].m_unAudioChunkID;
```

This list will be cleared in every sync frame and updated with new values.
```
if (m_bSyncFrameFlag) {
m_CAudioChunkNaviLits.ResetList();
}
```
So, we have to reset the uhdAudioChunkId in every sync frame.
Since we are extracting the unAudioChunkID in every sync frame and updating uhdAudioChunkId, we can avoid resetting uhdAudioChunkId explicitly.
 
Currently, the encoder supports only the Audio chunk type 0x1. So the chunk ID will be the same in all the frames.

As per the spec(section 4.2.4), the non-sync frame shouldn't be used as a random access point. So I am adding a logic to establish the synchronization only in the FTOC sync frames.
This will cause failure in the TsExtractorTest for seek test cases. The UHD test vector(sample_dts_uhd.ts) only has one FTOC sync frame. So after the seek operation, synchronization will not happen.

Reply:
Hi @rahulnmohan,

With your latest set of commits for this, the `TsExtractorTest` tests are failing for seek test cases, do you intend to fix this or should I do something about this? This bit is unclear and we cannot merge a PR with failing test cases right?

Reply:
Hi @rohitjoins, I will update the UHD test vector to have at least two sync frames.

Reply:
@rohitjoins 
I have updated the UHD test vector. Now the test stream contains three sync frames.

Reply:
> Since we are extracting the unAudioChunkID in every sync frame and updating uhdAudioChunkId, we can avoid resetting uhdAudioChunkId explicitly.

Does setting it explicitly breaks any tests? If not, I think we should do it so that we don't use a stale `uhdAudioChunkId` by mistake.

Reply:
Resetting the _uhdAudioChunkId_ won't break any tests. I will add the below code to the seek() function.
`uhdAudioChunkId.set(0)`